source/util/small_vector.h

```diff
@@ -461,14 +463,18 @@ class SmallVector {
   // The number of elements in |small_data_| that have been constructed.
   size_t size_;
 
-  // The pointed used to access the array of elements when the number of
-  // elements is small.
-  T* small_data_;
+  // A type with the same alignment and size as T, but will is POD.
+  struct alignas(T) PodType {
+    std::array<std::byte, sizeof(T)> data;
+  };
 
   // The actual data used to store the array elements.  It must never be used
   // directly, but must only be accessed through |small_data_|.
-  typename std::aligned_storage<sizeof(T), std::alignment_of<T>::value>::type
-      buffer[small_size];
+  std::array<PodType, small_size> buffer;
```

Comment:
this fails to compile on gcc 12.2:
``small_vector.h: In instantiation of ‘spvtools::utils::SmallVector<T, small_size>::SmallVector() [with T = const spvtools::opt::analysis::Type*; long unsigned int small_size = 8]’:
small_vector.h:51:42: error: member ‘spvtools::utils::SmallVector<const spvtools::opt::analysis::Type*, 8>::buffer’ is used uninitialized [-Werror=uninitialized]
   51 |         small_data_(reinterpret_cast<T*>(buffer.data())),`

```

Fixed by adding `= { }` at the end.

Reply:
I think this is fixed. I change the type of buffer to just a regular array instead of using std::array.